### DISCUSSION TITLE
Use target and source name in qualifiedByName method

### DISCUSSION THREAD

**Discussion by samsonsouley** [mapstruct/mapstruct#3250] [General]
I would like to return a value to the target base on the target name as the have the same source using the same qualifiedByName method.

```java
    @Mapper
    public interface MyMapper {
    @Mapping(target = "lowercaseName", source = "fullName", qualifiedByName = "changeName")
    @Mapping(target = "uppercaseName", source = "fullName", qualifiedByName = "changeName")
    void mapToDto(MyEntity entity);

    @Named("changeName")// Base on the target name i return uppercase or lowercase
     String changeName(String input) {
       if(target=="uppercaseName"){
          return source.toUpperCase();
        } 
        return source.toLowerCase();
    }
}
```

How can i do this ?
I want to avoid to create 2 method changeNameUpperCase and changeNameLowerCase which has almost the same content

**Comment by filiphr** [comment]
Currently you'll have to create 2 methods. We do not have a way to access the target property name. There is something coming up in 1.6, but I am not sure that it'll work for mapping methods

**Reply by samsonsouley** [reply]
Thnaks. It with really help if this is available.
